dump/mapping/bufr2ioda_satwnd_amv_goes.py

```diff
@@ -0,0 +1,309 @@
+
+    return container
+
+def create_obs_group(input_path, mapping_path, category, env):
+
+    comm = bufr.mpi.Comm(env["comm_name"])
+
+    description = _make_description(mapping_path, update=True)
+
+    # Check the cache for the data and return it if it exists
+    logging(comm, 'DEBUG', f'Check if bufr.DataCache exists? {bufr.DataCache.has(input_path, mapping_path)}')
+    if bufr.DataCache.has(input_path, mapping_path):
+        container = bufr.DataCache.get(input_path, mapping_path)
+        logging(comm, 'INFO', f'Encode {category} from cache')
+        data = iodaEncoder(description).encode(container)[(category,)]
+        logging(comm, 'INFO', f'Mark {category} as finished in the cache')
+        bufr.DataCache.mark_finished(input_path, mapping_path, [category])
+        logging(comm, 'INFO', f'Return the encoded data for {category}')
+        return data
+
+    container = _make_obs(comm, input_path, mapping_path)
+
+    # Gather data from all tasks into all tasks. Each task will have the complete record 
+    logging(comm, 'INFO', f'Gather data from all tasks into all tasks')
+    container.all_gather(comm)
+
+    logging(comm, 'INFO', f'Add container to cache')
+    # Add the container to the cache
+    bufr.DataCache.add(input_path, mapping_path, container.all_sub_categories(), container)
+
+    # Encode the data
+    logging(comm, 'INFO', f'Encode {category}')
+    data = iodaEncoder(description).encode(container)[(category,)]
+
+    logging(comm, 'INFO', f'Mark {category} as finished in the cache')
+    # Mark the data as finished in the cache
+    bufr.DataCache.mark_finished(input_path, mapping_path, [category])
+
+    logging(comm, 'INFO', f'Return the encoded data for {category}')
+    return data
+
+def create_obs_file(input_path, mapping_path, output_path):
+
+    comm = bufr.mpi.Comm("world")
+    container = _make_obs(comm, input_path, mapping_path)
+    container.gather(comm)
+
+    description = _make_description(mapping_path, update=True)
+
+    # Encode the data
+    if comm.rank() == 0:
+        netcdfEncoder(description).encode(container, output_path) 
+
+    logging(comm, 'INFO', f'Return the encoded data')
+
+
+if __name__ == '__main__':
+
+    start_time = time.time()
+
+    bufr.mpi.App(sys.argv)
+    comm = bufr.mpi.Comm("world")
+
+    # Required input arguments
+    parser = argparse.ArgumentParser()
+    parser.add_argument('-i', '--input', type=str, help='Input BUFR', required=True)
+    parser.add_argument('-m', '--mapping', type=str, help='BUFR2IODA Mapping File', required=True)
+    parser.add_argument('-o', '--output', type=str, help='Output NetCDF', required=True)
+
+    args = parser.parse_args()
+    mapping = args.mapping
+    infile = args.input
+    output = args.output
+
+    create_obs_file(infile, mapping, output)
+
+    end_time = time.time()
+    running_time = end_time - start_time
+    logging(comm, 'INFO', f'Total running time: {running_time}')
```

Comment:
Should we consider adding runtimes to the all subroutines in case we're curious if there are hang ups?